src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/P2PProtocolHandler.cs

```diff
@@ -225,6 +225,23 @@ private void HandleHello(HelloMessage hello)
                 $"capabilities: {string.Join(", ", capabilities)}");
         }
 
+        // Not sure if any client can do multiple same protocol at the same time, but Geth definitely only pick the latest version.
```

Comment:
Peers are choosing only one version of protocol at the start of connection - are sending `Hello` messages via `P2P` protocol with all available capabilities and choosing ones which are supported by both. If there are different versions (e.g. `eth` in version `66` and `67`), then always choosing higher one. Protocol version is not changing later.